src/Compute/Compute/Extension/SetAzureVMExtensionBaseCmdlet.cs

```
@@ -40,7 +40,7 @@ public class SetAzureVMExtensionBaseCmdlet : VirtualMachineExtensionBaseCmdlet
 
         [Alias("ExtensionName")]
         [Parameter(
-            Mandatory = false,
+            Mandatory = true,
```

Making a parameter from optional to required will cause  breaking change. Can you explain the reason?

---

stayed as optional

---

Looks it's still required.

---

my mistake.  The Name parameter, even though it was marked optional, was in fact mandatory. Users setting the extensions without extension name would have got an error from the API.

---

Please suppress exceptions from static analysis in https://github.com/Azure/azure-powershell/blob/main/tools/StaticAnalysis/Exceptions/Az.Compute/BreakingChangeIssues.csv